lib/collection/src/common/fetch_vectors.rs

```diff
@@ -105,17 +106,32 @@ impl ReferencedVectors {
 
     pub fn get(
         &self,
-        collection_name: &Option<&CollectionName>,
+        lookup_collection_name: &Option<&CollectionName>,
         point_id: PointIdType,
     ) -> Option<&Record> {
-        match collection_name {
+        match lookup_collection_name {
             None => self.default_mapping.get(&point_id),
             Some(collection) => {
                 let collection_mapping = self.collection_mapping.get(*collection)?;
                 collection_mapping.get(&point_id)
             }
         }
     }
+
+    pub fn convert_to_vectors_owned<'a>(
```

Comment:
It might be more comprehensive, if we just conver single vector at a time, and do the iteration outside of this function